source/Cosmos.Core/Memory/Heap.cs

```diff
@@ -44,6 +47,11 @@ public static unsafe void Init()
 		/// <returns>New pointer with specified size while maintaining old data.</returns>
         public static byte* Realloc(byte* aPtr, uint newSize)
 		{
+            if (mMemeoryGate != null)
```

Comment:
How can the mMemoryGate ever be null? Seems like we are doing useless checks

Reply:
we could replace it with ```mMemeoryGate?.Lock();``` it would do that same

Reply:
I would prefer if we dont do needless checks to reduce the performance cost, as these calls happen quite often